zppy/templates/coupled_global.py

```diff
@@ -841,8 +914,214 @@ def run(parameters, rgn):  # noqa: C901
+
+    def add_row(self, row_name):
+        """Add a row with the title name to the current group"""
+        if self.group is None:
+            raise RuntimeError("You must first insert a group with add_group()")
+        self.row = OutputRow(row_name, [])
+        if self.row not in self.cache[self.page][self.group]:
+            self.cache[self.page][self.group].append(self.row)
+        self.page.addRow(self.row, len(self.page.groups) - 1)  # type: ignore
+
+    def set_row(self, row_name):
+        """Sets the row with the title name as the current row"""
+        for output_row in self.cache[self.page][self.group]:
+            if row_name == output_row.title:
+                self.row = output_row
+                return
+        raise RuntimeError("There is no row titled: %s" % row_name)
+
+    def add_cols(self, cols):
+        """Add multiple string cols to the current row"""
+        self.row.columns.append(cols)  # type: ignore
+
+    def add_col(self, col, is_file=False, **kwargs):
+        """Add a single col to the current row. Set is_file to True if the col is a file path."""
+        if is_file:
+            self.row.columns.append(OutputFile(col, **kwargs))  # type: ignore
+        else:
+            self.row.columns.append(col)  # type: ignore
+
+    def generate_page(self):
+        """
+        Generate and return the location of the current HTML page.
+        """
+        self.index.toJSON(os.path.join(self.path, "index.json"))
+
+        default_mask = stat.S_IMODE(os.stat(self.path).st_mode)
+        rechmod(self.path, default_mask)
+
+        if os.access(self.path, os.W_OK):
+            default_mask = stat.S_IMODE(
+                os.stat(self.path).st_mode
+            )  # mode of files to be included
+            url = build_page(
+                self.page,
+                os.path.join(self.path, "index.json"),
+                default_mask=default_mask,
+            )
+            return url
+
+        raise RuntimeError("Error geneating the page.")
+
+    def generate_viewer(self):
+        """Generate the webpage"""
+        self.index.toJSON(os.path.join(self.path, "index.json"))
+
+        default_mask = stat.S_IMODE(os.stat(self.path).st_mode)
+        rechmod(self.path, default_mask)
+
+        if os.access(self.path, os.W_OK):
+            default_mask = stat.S_IMODE(
+                os.stat(self.path).st_mode
+            )  # mode of files to be included
+            build_viewer(
+                os.path.join(self.path, "index.json"),
+                diag_name="My Diagnostics",
+                default_mask=default_mask,
+            )
+
+
+class LandVariable(object):
+    def __init__(self, csv_row: List[str]):
+        # TODO: use these fields...
+        # the name of the ELM variable on the monthly h0 history file
+        self.variable_name = csv_row[0]
+        # “A” or “T” for global average over land area or global total, respectively
+        self.metric = csv_row[1]
```

Comment:
TODO: Actually use `A` or `T`. 

Going by @czender's comments https://github.com/E3SM-Project/zppy/pull/616#issuecomment-2305848885 and https://github.com/E3SM-Project/zppy/pull/616#issuecomment-2311408224, it seems like the logic change would belong more in `ts.py` than `coupled_global.py`. That would mean `ts.py` would _also_ have to parse the csv.

In any case, at some point, a decision on what to plot (`A` or `T`) must be made based on this csv. Currently, the csv parsing is done very late, after the plots themselves have been generated. This parsing will need to be moved much earlier in order to propagate the `A/T` information before it is needed.